src/main/java/com/bravo/user/model/filter/UserFilter.java

```diff
@@ -2,14 +2,18 @@
 
 import java.time.LocalDateTime;
 import java.util.Set;
+
+import lombok.Builder;
 import lombok.Data;
 
 @Data
+@Builder
 public class UserFilter {
 
   private Set<String> ids;
   private Set<String> firstNames;
   private Set<String> lastNames;
   private Set<String> middleNames;
+  private String name;
```

Comment:
Can you add the pros and cons of adding name here instead of its own class or trying to divide the name into its parts {firstName, lastName, MiddleName}.

Reply:
I've created a `UserNameFuzzyFilter` and `UserNameFuzzySpecification` to prevent inadvertent use of name fuzzy matching when exact name matching is desired.